WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -318,24 +238,32 @@ public void Synchronize()
 			});
 		}
 
-		private async Task<SyncInfo> GetSyncInfoAsync()
+		private HashSet<Script> FetchScripts(VerboseBlockInfo block)
 		{
-			var bcinfo = await RpcClient.GetBlockchainInfoAsync();
-			var pbcinfo = new SyncInfo(bcinfo);
-			return pbcinfo;
-		}
+			var scripts = new HashSet<Script>();
 
-		private void BlockNotifier_OnBlock(object sender, Block e)
-		{
-			try
-			{
-				// Run sync every time a block notification arrives. Synchronizer will stop when it finishes.
-				Synchronize();
-			}
-			catch (Exception ex)
+			foreach (var tx in block.Transactions)
 			{
-				Logger.LogError(ex);
+				for (int i = 0; i < tx.Outputs.Count; i++)
+				{
+					var scriptPubKey = tx.Outputs[i].ScriptPubKey;
+					if (scriptPubKey.IsScriptType(ScriptType.P2WPKH))
```

Comment:
Given the rpc json response contains the script type, we could avoid deserializing those scripts that are not p2wpkh.

Reply:
I suppose you are right but I don't think there will a noticeable perf gains

Reply:
I cannot know without measuring it. But we can be sure it is faster (probably not big gain) and reduce a lot of allocations and memory usage by sure.

Reply:
yeah I can try to measure defo worth a try.